app/components/Modals/BaseModal/BaseModal.jsx

```diff
@@ -30,11 +31,12 @@ const BaseModal = ({
   className,
   bodyClassName,
   style,
-  onAfterOpen
+  onAfterOpen,
+  shouldCloseWithEscapeKey
 }: Props) => (
   <ReactModal
     isOpen
-    onRequestClose={hideModal}
+    onRequestClose={() => shouldCloseWithEscapeKey ? hideModal() : undefined}
```

Comment:
:neckbeard: `shouldCloseWithEscapeKey && hideModal()` is a little more concise.

Reply:
nice will change that thanks @mhuggins